official/resnet/keras/keras_common.py

```diff
@@ -47,15 +55,21 @@ def __init__(self, batch_size):
     self._batch_size = batch_size
     super(TimeHistory, self).__init__()
     self.log_steps = 100
+    self.train_timestamp_history = []
 
   def on_train_begin(self, logs=None):
     self.record_batch = True
 
+  def on_train_end(self, logs=None):
+    self.train_finish_time = time.time()
+
   def on_batch_begin(self, batch, logs=None):
     if self.record_batch:
       self.start_time = time.time()
       self.record_batch = False
 
+    self.train_timestamp_history.append(BatchTimestamp(batch))
+
   def on_batch_end(self, batch, logs=None):
     if batch % self.log_steps == 0:
       elapsed_time = time.time() - self.start_time
```

Comment:
add self.train_timestamp_history.append(BatchTimestamp(batch)) here and pass in the time.time() used for elapsed_time so they are exactly the same.  This will ensure each entry in history will match logging.  Meaning if I calculate elapsed time with timestamp_history it will **exactly** match the log for those 100 steps.